plugins/module_utils/omevv_utils.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+
+# Dell OpenManage Ansible Modules
+# Version 9.8.0
+# Copyright (C) 2024 Dell Inc. or its subsidiaries. All Rights Reserved.
+
+# Redistribution and use in source and binary forms, with or without modification,
+# are permitted provided that the following conditions are met:
+
+#    * Redistributions of source code must retain the above copyright notice,
+#      this list of conditions and the following disclaimer.
+
+#    * Redistributions in binary form must reproduce the above copyright notice,
+#      this list of conditions and the following disclaimer in the documentation
+#      and/or other materials provided with the distribution.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED.
+# IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+# INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+# PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+# LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+# USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+
+from __future__ import (absolute_import, division, print_function)
+
+__metaclass__ = type
+
+VCENTER_INFO_URI = "/Consoles"
+PROFILE_URI = "/RepositoryProfiles"
+VCENTER_ERROR_MSG = "Unable to fetch the vCenter information."
+PROFILE_ERROR_MSG = "Unable to fetch the firmware respository profile information."
+
+
+class OMEVVINFO:
+    def __init__(self, omevv, module):
+        self.omevv = omevv
+        self.module = module
+
+    def get_all_vcenter_info(self):
+        """
+        Retrieves all the vCenter Information.
+
+        """
+        resp = self.omevv.invoke_request('GET', VCENTER_INFO_URI)
+        return resp
+
+    def get_firmware_repository_profile(self):
```

Comment:
Good to move this to anew util with the name omevv_firmware_api and add all firmware related API's into it.

Also Enhance this method to get a particular profile based on name, id as required.

Reply:
Do proper naming of functions, if it fetches all profile, make sure function name is clear, same for individual profile name fetching functions

Reply:
Function is changed to handle both scenarios.